openbb_platform/providers/intrinio/openbb_intrinio/models/etf_search.py

```diff
@@ -0,0 +1,140 @@
+
+from openbb_core.provider.abstract.fetcher import Fetcher
+from openbb_core.provider.standard_models.etf_search import (
+    EtfSearchData,
+    EtfSearchQueryParams,
+)
+from openbb_core.provider.utils.errors import EmptyDataError
+from openbb_core.provider.utils.helpers import (
+    ClientResponse,
+    ClientSession,
+    amake_request,
+)
+from openbb_intrinio.utils.references import ETF_EXCHANGES
+from pandas import DataFrame
+from pydantic import Field
+
+
+class IntrinioEtfSearchQueryParams(EtfSearchQueryParams):
+    """
+    Intrinio ETF Search Query Params.
+
+    Source: https://docs.intrinio.com/documentation/web_api/search_etfs_v2
+    """
+
+    exchange: Union[None, ETF_EXCHANGES] = Field(
+        default=None,
+        description="Target a specific exchange by providing the MIC code.",
+    )
+
+
+class IntrinioEtfSearchData(EtfSearchData):
+    """Intrinio ETF Search Data."""
+
+    __alias_dict__ = {
+        "intrinio_id": "id",
+        "symbol": "ticker",
+        "exchange": "exchange_mic",
+    }
+
+    exchange: Optional[str] = Field(
+        default=None,
+        description="The exchange MIC code.",
+    )
+    figi_ticker: Optional[str] = Field(
+        None,
+        description="The OpenFIGI ticker.",
+    )
+    ric: Optional[str] = Field(
+        None,
+        description="The Reuters Instrument Code.",
+    )
+    isin: Optional[str] = Field(
+        None,
+        description="The International Securities Identification Number.",
+    )
+    sedol: Optional[str] = Field(
+        None,
+        description="The Stock Exchange Daily Official List.",
+    )
+    intrinio_id: Optional[str] = Field(
+        None,
+        description="The unique Intrinio ID for the security.",
+    )
+
+
+class IntrinioEtfSearchFetcher(
+    Fetcher[IntrinioEtfSearchQueryParams, List[IntrinioEtfSearchData]]
+):
+    """Intrinio ETF Search Fetcher."""
+
+    @staticmethod
+    def transform_query(params: Dict[str, Any]) -> IntrinioEtfSearchQueryParams:
+        """Transform query."""
+        return IntrinioEtfSearchQueryParams(**params)
+
+    @staticmethod
+    async def aextract_data(
+        query: IntrinioEtfSearchQueryParams,
+        credentials: Optional[Dict[str, str]],
+        **kwargs: Any,
+    ) -> List[Dict]:
+        """Return the raw data from the Intrinio endpoint."""
+
+        api_key = credentials.get("intrinio_api_key") if credentials else ""
+        BASE = "https://api-v2.intrinio.com/etfs"
+        if query.exchange is not None:
+            url = f"{BASE}?exchange={query.exchange.upper()}&page_size=10000&api_key={api_key}"
+        if not query.query and query.exchange is None:
+            url = f"{BASE}?page_size=10000&api_key={api_key}"
+        if query.query and query.exchange is None:
+            url = f"{BASE}/search?query={query.query}&page_size=10000&api_key={api_key}"
+
+        data = []
+
+        async def response_callback(response: ClientResponse, session: ClientSession):
```

Comment:
I'd suggest this and it should fix the typing too:

```
async def response_callback(response: ClientResponse, session: ClientSession) -> List[Dict]:
        """Async response callback."""
        results = await response.json()

        if "messages" in results:
            messages = results["messages"]
            raise RuntimeError(str(messages))
        etfs = results.get("etfs", [])
        if etfs:
            data.extend(etfs)
            while "next_page" in results:
                next_page = results["next_page"]
                next_url = f"{url}&next_page={next_page}"
                results = await amake_request(next_url, session=session, **kwargs)
                etfs = results.get("etfs", [])
                if etfs:
                    data.extend(etfs)
        return data
```

Reply:
The typing is because the response from the HTTP client is ambiguous.  Calling "results" a 'dict' just changes the typing error to another form.

![Screenshot 2024-03-13 at 8 15 06 AM](https://github.com/OpenBB-finance/OpenBBTerminal/assets/85772166/3bb2b67f-fbe6-4198-accd-1bd2da690a3a)